internal/bot/bot.go

```diff
@@ -358,6 +377,20 @@ func (b *Bot) getProfitOrLossText(priceChangePercentage float64) string {
 	return profitOrLossText
 }
 
+// updateVolumeTraded fetches the volume traded of all coins from the market and stores them in the CoinVolumes map.
+func (b *Bot) updateVolumeTraded(ctx context.Context) error {
+	b.botLog.Debug("Fetching volume traded of all coins.")
+
+	volumeTraded, err := b.market.GetCoinsVolume(ctx)
+	if err != nil {
+		return err
+	}
+
+	market.CoinVolumes = volumeTraded
```

Comment:
I feel like the volumes shouldn't be stored in a global variable `CoinVolumes` because they are different per supported marketplace. And prefixing them like `BinanceCoinVolumes` would be ugly... 

Perhaps we should store this information per bot instance instead (i.e. within the `Bot` struct) using a multi-dimensional map like `map[string]CoinVolumeTradedMap`. The market name can be retrieved with `b.market.Name()`. This should make it more future proof for when other marketplace(s) are added.

Lmk if you got another idea.

Reply:
I see what you are saying, we might be running multiple instances of the bot in parallel, on different exchanges. I think your suggestion makes a lot of sense, I'm just not exactly sure how I should change this. I tried to push a refactor with https://github.com/sleeyax/voltra/pull/16/commits/95b23101615eb9f059851db2c7fd931ee163b29a, let me know if this is ok

Reply:
Hmm what I initially had in mind is to replace `type CoinVolumeTradedMap map[string]float64` with something like this:

```go
// TradeVolumes is a map of coin symbols to their 24h quote asset volume traded.
type TradeVolumes map[string]float64

// TradeVolumesByMarket is a map of market names to their trade volumes.
type TradeVolumesByMarket map[string]TradeVolumes
```

And then add `coinVolumes TradeVolumesByMarket` to the `Bot` struct instead.

**But** thinking about it this I'm probably thinking way ahead if you take the current iteration of the project into consideration. Ultimately it depends on how we end up handling multiple exchanges in the future; You could have multiple bot instances like you mentioned or just one which takes in a slice (array) of markets and manages them internally, spawning go routines etc. as required. I was thinking in terms of the latter but that doesn't fit right now.

So your changes are fine.